packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -164,12 +165,32 @@ class _DefaultPub implements Pub {
     bool offline = false,
     bool generateSyntheticPackage = false,
     String flutterRootOverride,
+    bool check = false,
   }) async {
     directory ??= _fileSystem.currentDirectory.path;
     final File packageConfigFile = _fileSystem.file(
       _fileSystem.path.join(directory, '.dart_tool', 'package_config.json'));
     final Directory generatedDirectory = _fileSystem.directory(
       _fileSystem.path.join(directory, '.dart_tool', 'flutter_gen'));
+    final File lastVersion = _fileSystem.file(
+      _fileSystem.path.join(directory, '.dart_tool', 'version'));
+    final File currentVersion = _fileSystem.file(
+      _fileSystem.path.join(Cache.flutterRoot, 'version'));
+    final File pubspecYaml = _fileSystem.file(
+      _fileSystem.path.join(directory, 'pubspec.yaml'));
+
+    // If the pubspec.yaml is older than the package config file and the last
+    // flutter version used is the same as the current version. This will incorrectly
+    // skip pub on the master branch if dependencies are being added/removed from the
+    // flutter framework packages.
```

Comment:
Should we add a note about that there or file an issue?

Reply:
practically speaking, I don't think we'll ever hit this condition. But if someone does this is a nice breadcrumb.

Reply:
The change has to be uncommitted too, because as soon as you make a commit the flutter version changes